dsc/src/util.rs

```diff
@@ -288,3 +289,61 @@ pub fn enable_tracing(trace_level: &TraceLevel, trace_format: &TraceFormat) {
         eprintln!("Unable to set global default tracing subscriber.  Tracing is diabled.");
     }
 }
+
+pub fn parse_input_to_json(value: &str) -> String {
+    match serde_json::from_str(value) {
+        Ok(json) => json,
+        Err(_) => {
+            match serde_yaml::from_str::<Value>(value) {
+                Ok(yaml) => {
+                    match serde_json::to_value(yaml) {
+                        Ok(json) => json.to_string(),
+                        Err(err) => {
+                            error!("Error: Failed to convert YAML to JSON: {err}");
+                            exit(EXIT_DSC_ERROR);
+                        }
+                    }
+                },
+                Err(err) => {
+                    error!("Error: Input is not valid JSON or YAML: {err}");
+                    exit(EXIT_INVALID_INPUT);
+                }
+            }
+        }
+    }
+}
+
+pub fn get_input(input: &Option<String>, stdin: &Option<String>, path: &Option<String>) -> String {
+    let value = match (input, stdin, path) {
+        (Some(_), Some(_), None) | (None, Some(_), Some(_)) => {
+            error!("Error: Cannot specify both stdin and --input or --path");
+            exit(EXIT_INVALID_ARGS);
+        },
+        (Some(input), None, None) => input.clone(),
+        (None, Some(stdin), None) => stdin.clone(),
+        (None, None, Some(path)) => {
+            info!("Reading input from file {}", path);
+            match std::fs::read_to_string(path) {
+                Ok(input) => input.clone(),
+                Err(err) => {
+                    error!("Error: Failed to read input file: {err}");
+                    exit(EXIT_INVALID_INPUT);
+                }
+            }
+        },
+        (None, None, None) => {
+            return String::new();
+        },
+        _default => {
+            /* clap should handle these cases via conflicts_with so this should not get reached */
+            error!("Error: Invalid input");
+            exit(EXIT_INVALID_ARGS);
+        }
+    };
+
+    if value.is_empty() {
```

Comment:
Unclear to me when this value is empty?  Maybe add a comment.  Should also have test cases when any of the inputs is empty string.

Reply:
I'm not completely sure either - I put it in because the existing dsc_args.tests.ps1 were failing on the CI but not locally within `get_inputs` - it seems like it was detecting `Some` for both `stdin` and either `document` or `path` but `stdin` was actually empty.

Reply:
I think you can technically pass an empty string via stdin: "" | dsc, although not entirely sure